sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFilters.scala

```diff
@@ -725,26 +757,8 @@ class ParquetFilters(
 
       case sources.In(name, values) if pushDownInFilterThreshold > 0 && values.nonEmpty &&
           canMakeFilterOn(name, values.head) =>
-        val fieldType = nameToParquetField(name).fieldType
-        val fieldNames = nameToParquetField(name).fieldNames
-        if (values.length <= pushDownInFilterThreshold) {
-          values.distinct.flatMap { v =>
-            makeEq.lift(fieldType).map(_(fieldNames, v))
-          }.reduceLeftOption(FilterApi.or)
```

Comment:
It seems a regression, For example:
```
In(id, 1, 3, 5, 100000)
```

Before this PR, the pushded predicate is: id = 1 or id = 3 or id = 5 or id = 100000,
the current pushded predicate is: id >= 1 and id <= 100000

Reply:
Could you add a test coverage for @wangyum 's case, please, @huaxingao ?

Reply:
I will change back to use equal Or equal if the number of values in the `In` set <=  `pushDownInFilterThreshold`. Only use parquet `In` predicate if the number of values in the `In` set >  `pushDownInFilterThreshold`. The existing tests coverage should be good, just need to change the expected type back to equal Or equal when the number of values in the `In` set <=  `pushDownInFilterThreshold`.